components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/DefaultOIDCSessionStateManager.java

```diff
@@ -109,7 +110,19 @@ public Cookie addOPBrowserStateCookie(HttpServletResponse response, HttpServletR
             removeOPBrowserStateCookiesInRoot(request, response);
 
             cookie = new ServletCookie(OIDCSessionConstants.OPBS_COOKIE_ID, opbsValue);
-            if (isOrganizationQualifiedRequest()) {
+            String appResidentOrgId = PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                    .getApplicationResidentOrganizationId();
+            if (StringUtils.isNotBlank(appResidentOrgId)) {
+                // Handling the cookie path for request coming with the path `/t/<tenant-domain>/o/<org-id>`.
+                // The cookie will be set to the root tenant path.
+                loginTenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
```

Comment:
Shall we get the loginTenantDomain  from the `OAuth2Util.extractTenantDomain`  method and simply these logics?